packit/upstream.py

```diff
@@ -921,3 +919,58 @@ def get_version_from_tag(self, tag: str) -> str:
             )
             logger.error(msg)
             raise PackitException(msg)
+
+    def get_inner_archive_dir(self, archive: str) -> str:
+        """
+        Returns tar archive top-level directory
+
+        Currently supported archives:
+        * tar including compression (for details check python tarfile module doc)
+
+        :param archive: name of archive
+        :return: archive top level directory
+        """
+        if tarfile.is_tarfile(f"{self.absolute_specfile_dir}/{archive}"):
+            logger.debug(f"Archive {archive} is tar.")
+            return self.get_inner_tar_dir(archive)
+
+        raise PackitException("Unable to identify archive type.")
```

Comment:
I can tell you right away there is a ton of more different archive types than just tar and wouldn't be wise to halt on such a thing: so instead please let's do `return None` here and do `        new_setup_line += f" -n {self.package_config.upstream_package_name}-{version}"` if that happens

Reply:
:+1: on returning `None` in case of unknown archive types.

In that case, let's try the config value and then the old behaviour. (As we agreed in the morning.)

Reply:
done